api/src/services/user-db.js

```diff
@@ -15,17 +15,20 @@ const DB_NAME_BLOCKED_CHARS = /[^a-z0-9_$()+/-]/g;
 // across node versions: https://github.com/nodejs/node/issues/20355
 // We are currently testing the exact content of the map function in tests/unit/lib/user-db.js
 /* eslint-disable no-var */
+/* istanbul ignore next */
+
 const readMapFunction = function (doc) {
-  var parts = doc._id.split(':');
+  var parts = doc._id.split(':');/*NOSONAR*/
```

Comment:
This is horrible. How can I disable sonar for a whole section?

Reply:
The other option is to [update the repo config](https://docs.communityhealthtoolkit.org/contribute/code/static-analysis/#ignoring-all-rules-for-a-file) to exclude the whole file, either from all Sonar linting, or just exclude it from particular rules...

Reply:
Alternatively, it looks like we can configure a custom pattern to match on that would allow for [excluding blocks of code](https://docs.sonarsource.com/sonarqube/latest/project-administration/analysis-scope/#ignoring-blocks-within-files)

Just add this to the `.sonarcloud.properties`:

```
sonar.issue.ignore.block=e1
sonar.issue.ignore.block.e1.beginBlockRegexp=NOSONAR_BEGIN
sonar.issue.ignore.block.e1.endBlockRegexp=NOSONAR_END
```

Then put `//NOSONAR_BEGIN` before the block to ignore and `//NOSONAR_END` after the block.

Reply:
Cool, I like this. I didn't know you can add sonar props! Thanks for the research!